src/libsyntax/util/small_vector.rs

```diff
@@ -113,6 +156,15 @@ impl<T> SmallVector<T> {
         }
     }
 
+
+    pub fn get_mut(&mut self, idx: usize) -> Option<&mut T> {
+        match self.repr {
+            One(ref mut v) if idx == 0 => Some(v),
+            Many(ref mut vs) => vs.get_mut(idx),
+            _ => None
```

Comment:
get_mut should have the same semantics as get. For returning an Option, you should have another method (maybe_get_mut or something).